File: modules/liferay-sample-module/liferay-sample-module-js-web/src/app/sample/service/sample.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { Sample, SamplePaginated } from '../model/sample';

declare const Liferay: any;

@Injectable({
  providedIn: 'root',
})
export class SampleService {
  constructor(private _httpClient: HttpClient) {}

  url = (<any>window).SampleWorkspace.baseUrl + '/samples';

  getSamples(): Observable<SamplePaginated> {
    return this._httpClient.get<SamplePaginated>(this.url);
  }

  getSample(id: number): Observable<Sample> {
    return this._httpClient.get<Sample>(`${this.url}/${id}`);
  }

  createSample(name: string): Observable<Sample> {
    return this._httpClient.post<Sample>(this.url, {
      name,
    });
  }

  updateSample(sample: Sample): Observable<void> {
    return this._httpClient.put<void>(this.url, sample);
  }

  deleteSample(id: number): Observable<void> {
    return this._httpClient.delete<void>(`${this.url}/${id}`);
  }
}

File: modules/liferay-sample-module/liferay-sample-module-api/src/main/java/com/liferay/sample/module/service/SampleService.java
package com.liferay.sample.module.service;

import com.liferay.sample.module.model.SampleObject;

import java.util.List;

/**
 * @author Rafael Oliveira
 */
public interface SampleService {

	public SampleObject addSample(SampleObject sampleObject);

	public void deleteSample(String id);

	public SampleObject getSample(String id);

	public List<SampleObject> getSamples();

	public SampleObject updateSample(String id, SampleObject sampleObject);

}